### DISCUSSION TITLE
Category and page route is the same how to deal with it in next.js?

### DISCUSSION THREAD

**Discussion by stevemarksd** [vercel/next.js#34901] [Help]
For example my category is: `/home-and-garden/home/` and my post can be `/home/how-to-do-something/`. I cannot change the post slug because are inherited from old website and sometimes certain slugs must be this way. Same for category.Would be nice in next.js to have something like `getRouteFromSlug`, does it exist or what solution do you recommend?

I could have an if/else and then show the correct component, but no code splitting right?

**Comment by tpetersons** [comment]
Can you introduce resource name before this path?
`categories/home-and-garden/home/`
`posts/home/how-to-do-something/`
This still doesn't comply with best practices designing URLs, but it might be better than if / else statement in the code.

**Reply by stevemarksd** [reply]
Thanks, but yeah, thats the easy solution but won't work in this case. I ended up with an if/else with dynamic import to split files. Not nice, but the only thing that works.
